### ISSUE TITLE
Completed commands (eg git push) sometimes fail to close block on fish shell

### ISSUE THREAD

**Issue by daogilvie** [warpdotdev/Warp#2013]
### Describe the bug

When running commands (most frequently `git push`) I am finding that Warp sometimes does not successfully detect that the program has completed and that a new prompt+block is needed.
The cursor is simply stuck on an empty line at the end of a current block, and I need to press Ctrl+C and then Enter to move to a new block.

I have disabled my custom prompt, and am using the fish shell, so this may merit another entry in the compatibility pages.

### To Reproduce

Honestly I have not managed to find precise, reliable replication. If I discover what is happening I will update this issue.

### Expected behaviour

When the command (eg `git push`) has completed, I expect to see a new block and prompt, awaiting input.

### Screenshots

~~Most of the times I find this day-to-day are not safely screenshareable. I will keep trying to get a suitable grab.~~
This is what happens — I've left this for up to an hour previously, and nothing changes unless I CTRL+C -> Enter as described above.
<img width="1124" alt="Screenshot 2022-10-27 at 11 21 57" src="https://user-images.githubusercontent.com/2018130/198260283-c7952aeb-5c9d-4565-b05b-97f9c4b265a1.png">

### Operating System

MacOS

### OS Version

12.6.1

### Shell Version

fish 3.5.1

### Warp Version

v0.2022.10.18.08.10.stable_03

### Additional context

I have previously discussed a related issue over an email exchange.

### Does this block you from using Warp daily?

No

**Comment by dannyneira** [comment]
Thanks for letting us know! We're sorry about this, this is not an ideal experience.

**Comment by dannyneira** [comment]
My guess is that you normally have a [custom prompt, theme, or plugin and Warp is unable to parse it.](https://docs.warp.dev/features/prompt#custom-prompt-compatibility-table)

You can check whether it’s something in your dotfiles by [setting up clean configs:](https://docs.warp.dev/help/known-issues#configuring-and-debugging-your-rc-files)
Run `echo 'ZDOTDIR=/' > ~/.zshenv`
This forces Zsh to run with zero configs.

You can disable parts of your dotfiles just for Warp by using this conditional statement:

```sh

if [[ $TERM_PROGRAM != "WarpTerminal" ]]; then
##### WHAT YOU WANT TO DISABLE FOR WARP - BELOW

# Whatever you want to disable

##### WHAT YOU WANT TO DISABLE FOR WARP - ABOVE
fi

```

We have a list of incompatible tools here: <https://docs.warp.dev/help/known-issues#list-of-incompatible-tools>
And more info on custom prompts here: <https://docs.warp.dev/features/prompt>

**Comment by daogilvie** [comment]
Hi @dannyneira — you actually have already given me this advice but I've not been able to narrow down what fish config might be causing the issue. I'll attempt to use warp all next week with literally no config outside of defaults and see if the problem recurs at all.

On a side note, running with various configurations has shown me that warp doesn't handle fish's [fish_mode_prompt](file:///usr/local/Cellar/fish/3.5.1/share/doc/fish/cmds/fish_mode_prompt.html?highlight=fish_mode_prompt) behaviour well, so you may want to add that to the list of incompatible tools! Warp inserts the output of that function into the input, rather than ignoring it as part of the prompt, so if a fish user like myself has VI bindings enabled, they have to replace the default function.

**Comment by dannyneira** [comment]
hi @daogilvie any updated on the testing you mentioned above?

**Comment by daogilvie** [comment]
Hi @dannyneira, I found that this did not stop the occasional failed block — I also found that unfortunately a handful of TUI applications I used regularly had a worse experience in warp (K9S is an example) and so I went back to iterm2 for the moment. I have also recently had a machine upgrade, so haven't had warp installed for the last few weeks.
I'm sorry to leave the hanging issue open — I'll close it now, and will make a note to return to re-try warp in the next few days/weeks. Thanks!
